src/Components/Header.jsx

```diff
@@ -0,0 +1,296 @@
+import { Box, Flex, Text, Tooltip } from "@sparrowengg/twigs-react";
+import { Input } from "@sparrowengg/twigs-react";
+
+import { SearchIcon } from "./Icons";
+
+import DownArrow from "./../asset/downArrow.svg";
+import DownArrowWhite from "./../asset/downArrowWhite.svg";
+import Profile from "./../asset/profile.png";
+import Sun from "./../asset/sun.svg";
+import Moon from "./../asset/moon.svg";
+
+import { useDispatch, useSelector } from "react-redux";
+
+import { DarkMode, LightMode } from "../redux/light-dark/lightDarkTypes";
+
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSub,
+  DropdownMenuSeparator,
+  DropdownMenuSubContent,
+  DropdownMenuSubTrigger,
+} from "@sparrowengg/twigs-react";
+import { useState } from "react";
+import { Link, useLocation } from "react-router-dom";
+
+const Header = () => {
+  const [dataSearch, setDataSearch] = useState([]);
+  const [searchText, setSearchText] = useState("");
+
+  const location = useLocation();
+
+  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+
+  const mode = useSelector((store) => store.lightdarkmode.lightDarkMode);
+  const data = useSelector((store) => store.fakeapidata.data);
+
+  const dispatch = useDispatch();
+
+  // function to handle the light and dark mode
+  const handleModeChange = () => {
+    dispatch({
+      type: mode === LightMode ? DarkMode : LightMode,
+    });
+  };
+
+  const getSearchData = (searchVal) => {
+    let newData = data.filter((item) => item.title.includes(searchVal));
+
+    setDataSearch(newData);
+  };
+
+  return (
+    <Flex
+      css={{
+        width: "100%",
+        backgroundColor: "$primary",
+        padding: "16px 30px",
+      }}
+      alignItems="center"
+      justifyContent="space-between"
+    >
+      {/* header left section */}
+      <Flex
+        alignItems="center"
+        gap="40px"
+        css={{
+          width: "100%",
+        }}
+      >
+        <Flex
+          justifyContent="flex-start"
+          gap="40px"
+          css={{
+            width: "auto",
+          }}
+        >
+          <Link
+            to={"/"}
+            style={{
+              textDecoration: "none",
+            }}
+          >
+            <Text
+              css={{
+                fontSize: "$md",
+                fontWeight: 600,
+                color:
+                  location.pathname === "/" ? "$textPrimary" : "$textSecondary",
+                cursor: "pointer",
+              }}
+            >
+              Dashboard
+            </Text>
+          </Link>
+
+          <Link
+            to={"/advancequarry"}
+            style={{
+              textDecoration: "none",
+            }}
+          >
+            <Text
+              css={{
+                fontSize: "$md",
+                cursor: "pointer",
+                fontWeight: 600,
```

Comment:
@9xVibee can you remove these fontSizes and fontWeight from styles, can you check the entire pr